### ISSUE TITLE
Interactive broker connectivity

### ISSUE THREAD

**Issue by baeriswyld** [ccxt/ccxt#6508]
Hello !

Any chance you consider integrated non-crypto exchanges ?
We use your CCXT template to integrate Interactive Brokers (www.interactivebrokers.co.uk/).

We could potential make this open source in collaboration with you. 
Let me know !

Best wishes,
Daniel

**Comment by frosty00** [comment]
Hi there,

> Any chance you consider integrating non-crypto exchanges ?

I don't see why not, as long it supports the same data format for `fetchTickers / fetchMarkets / etc`

> We could potential make this open source in collaboration with you.

If you do you stand a chance at getting it merged ) We are a free software project with a very limited budget. Make a PR and I'll take a look!

Cheers.

**Comment by kroitor** [comment]
@baeriswyld hey)

I suppose it's this API: https://interactivebrokers.com/api/doc.html. It would be great if you could submit a PR – we will do our best to assist on it and get it merged. If not, we will get to it as soon as we can.

**Comment by baeriswyld** [comment]
Hi thanks for considering!

We are trying with this library but strongly slightly: https://github.com/erdewit/ib_insync

Will let you know about our progress.

Cheers,
Daniel

**Comment by kroitor** [comment]
@baeriswyld awesome! Thx for your involvement!

**Comment by omidkrad** [comment]
Hi @baeriswyld, this would be great. However, CCXT needs it to be implemented in JavaScript and transpilable to Python/PHP.

**Comment by omidkrad** [comment]
Other exchanges of interest:
- [Alpaca](https://alpaca.markets)
- [Tradier](https://brokerage.tradier.com)
- [DriveWealth](https://drivewealth.com)

**Comment by magick93** [comment]
For Interactive Brokers, there is this javascript library - https://github.com/pilwon/node-ib

**Comment by baeriswyld** [comment]
Hello guys,

we gave up on implementing Interactive Brokers (IB) with Python. According to our developer, the IB API is utterly badly documented. 

We have almost finished the integration of [SAXO API](https://www.developer.saxo/openapi/learn). But warning Live Trading with API on Saxo is only available with min. deposit of 100kUSD. However, great API documentation and support, and can be used for free as trading simulation. Once we are done with it, we can potentially share it here.

If you just want to implement a paper trading solution, the API of [Alpaca](https://alpaca.markets/docs/api-documentation/api-v2/) might be the perfect solution. Alpaca live trading is only available for US domiciles.

**Comment by baeriswyld** [comment]
> Hi @baeriswyld, this would be great. However, CCXT needs it to be implemented in JavaScript and transpilable to Python/PHP.

Hello,
could you explain me more about that? Are you free at some point to have a chat ? Can I reach you somehow on Slack ?

**Comment by baeriswyld** [comment]
> For Interactive Brokers, there is this javascript library - https://github.com/pilwon/node-ib

Is it possible to integrate this into CCXT ?

**Comment by kroitor** [comment]
@baeriswyld yes, but because it's mostly a WS API, it does not belong to CCXT, since CCXT is a RESTful lib. CCXT Pro adds the WS support to it, therefore this API can go into CCXT Pro: #6543

**Comment by kroitor** [comment]
@baeriswyld 

> CCXT needs it to be implemented in JavaScript and transpilable to Python/PHP.
> Hello, could you explain me more about this? 

This is covered in the CONTRIBUTING doc which is a _"must read first thing"_ for all code contributors: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support

> Are you free at some point to have a chat ? Can I reach you somehow on Slack ?

You're welcome ) 

 ↓

<sub><sub>[![Discord](https://img.shields.io/discord/690203284119617602?logo=discord&logoColor=white)](https://discord.gg/dhzSKYU)</sub></sub>

**Comment by kimasplund** [comment]
Maybe reach out to some of the none PDT rule brokers? Like zerotrade etc.
I would think a bot would trip the pdt daytrade rule nd get you banned for just about everyone thinking of giving it a try.

**Comment by amalysh** [comment]
Hello guys,
is someone already on implementing it? I'm asking because we consider to start implementation for IBB.

**Comment by lomagno2003** [comment]
Hello everyone, I'm also looking to use IB with Freqtrade (which uses CCXT) and wanted to check if someone is working on such integration.

Cheers!

**Comment by roozich** [comment]
Hi, is there any progress for IB exchange? I saw a draft but it looks inactive
![image](https://github.com/ccxt/ccxt/assets/1100023/2da4bdc1-6f38-450b-96d8-33bc05e15f01)

**Comment by prononext** [comment]
how is it going here? Anyone still on it?
